core/domain/user_query_domain.py

```diff
@@ -19,26 +19,43 @@
 from __future__ import annotations
 
 import collections
+import datetime
 
 from core import feconf
 from core import utils
 from core.constants import constants
 
-attribute_names = [ # pylint: disable=invalid-name
-    predicate['backend_attr'] for predicate in (
-        constants.EMAIL_DASHBOARD_PREDICATE_DEFINITION)]
+from typing import List, Optional
 
-UserQueryParams = collections.namedtuple( # pylint: disable=invalid-name
-    'UserQueryParams', attribute_names, defaults=(None,) * len(attribute_names))
+attribute_names = [ # pylint: disable=invalid-name
+        predicate['backend_attr'] for predicate in (
+            constants.EMAIL_DASHBOARD_PREDICATE_DEFINITION)]
+
+# We can't inject dynamic data(attribute_names) into namedtuple(UserQueryParams)
+# without raising mypy error. Therefore , we need to silence these errors
+# explicitly using # type: ignore[misc] for now.
+# For more information : https://github.com/python/mypy/issues/848
+UserQueryParams = collections.namedtuple( # type: ignore[misc] # pylint: disable=invalid-name
+        'UserQueryParams',
+        attribute_names,
+        defaults=(None,) * len(attribute_names) # type: ignore[misc]
+        )
```

Comment:
Hmm, for type safety, can we redefine attribute names in this file and just check they match the stuff in constants?

Reply:
Maybe we can but let's suppose if our data grows in constants we will need to scale our redefined attributes here to match those in constants .

Reply:
ok , let me try to implement that !!